core/templates/pages/exploration-player-page/services/command-executor.service.ts

```diff
@@ -0,0 +1,210 @@
+      return;
+    }
+    if (this.hostname === '') {
+      this.cachedOuterFrameMessage = stateToCommand[id];
+    } else {
+      this.windowWrapperMessageService.postMessageToParent(stateToCommand[id],
+        this.hostname);
+    }
+  }
+
+  continueClick(windowRef) {
+    try {
+      var button = windowRef.nativeWindow.document.getElementsByClassName(
+        'oppia-learner-confirm-button'
+      )[0] as HTMLElement;
+      button.click();
+    } catch {
+      var button = windowRef.nativeWindow.document.getElementsByClassName(
+        'protractor-test-next-card-button')[0] as HTMLElement;
+      button.click();
+    }
+    this.setElementsOnPage = 0;
+  }
+
+  fillTextBox(windowRef, text) {
+    var box = windowRef.nativeWindow.document.getElementsByClassName(
+      'form-control')[0];
+    box.value = text;
+    var evt = document.createEvent('HTMLEvents');
+    evt.initEvent('change', false, true);
+    box.dispatchEvent(evt);
+    var button = windowRef.nativeWindow.document.getElementsByClassName(
+      'oppia-learner-confirm-button')[0] as HTMLElement;
+    button.click();
+  }
+
+  addSet(windowRef, elements) {
+    if (!this.setElementsOnPage) {
+      this.setElementsOnPage = 0;
+    }
+    for (var i = 0; i < elements.length; i++) {
+      if (this.setElementsOnPage === 0) {
+        var box = windowRef.nativeWindow.document.querySelector(
+          '.form-control') as HTMLInputElement;
+        box.value = elements[i];
+        var evt = document.createEvent('HTMLEvents');
+        evt.initEvent('change', false, true);
+        box.dispatchEvent(evt);
+        this.setElementsOnPage += 1;
+      } else {
+        var addButton = windowRef.nativeWindow.document.querySelectorAll(
+          '.btn-secondary'
+        )[0] as HTMLElement;
+        addButton.click();
+        this.setElementsOnPage += 1;
+        var box = windowRef.nativeWindow.document.querySelectorAll(
+          '.form-control')[this.setElementsOnPage - 1] as HTMLInputElement;
+        box.value = elements[i];
+        var evt = document.createEvent('HTMLEvents');
+        evt.initEvent('change', false, true);
+        box.dispatchEvent(evt);
+      }
+    }
+  }
+
+  removeSet(windowRef, element) {
+    var boxes = windowRef.nativeWindow.document.querySelectorAll(
+      '.form-control');
+    for (var i = 0; i < boxes.length; i++) {
+      if (boxes[i].value === element) {
+        var deleteButton =
+        windowRef.nativeWindow.document.querySelectorAll(
+          '.oppia-delete-list-entry-button')[i];
+        deleteButton.click();
+      }
+    }
+    this.setElementsOnPage -= 1;
+  }
+
+  submit(windowRef) {
+    var button = windowRef.nativeWindow.document.getElementsByClassName(
+      'oppia-learner-confirm-button')[0] as HTMLElement;
+    button.click();
+    this.setElementsOnPage = 0;
+  }
+
+  enterFraction(windowRef, fraction) {
+    var fractionElementName =
+    'form-control ng-valid-f-r-a-c-t-i-o-n_-f-o-r-m-a-t_-e-r-r-o-r';
```

Comment:
We shouldn’t be depending on Angular generated classes since those could change in the future. Prefer specific classes to the operations being done here.